esm/html/document.js

```diff
@@ -102,3 +103,53 @@ export class HTMLDocument extends Document {
     return element;
   }
 }
+
+/** @typedef {{ [WINDOW] : Object }} Context */
+/**
+ * @constructor
+ * @param {Context} context
+ */
+export class HTMLDocument extends _HTMLDocument {
+
+  constructor (context = {}) {
+    super();
+    const useState = (target, name) => {
+      return ((_0, _1={}) => {
+        const g = (name in _1) ? _1 : _0;
+        return {
+          get: () => g[name],
+          set: (v) => (g[name] = v) || true,
+        }
+      })(target, context[WINDOW]);
+    };
+    this[WINDOW] = {
+      set: (target, name, value) => {
+        return useState(target, name).set(value);
+      },
+      get: (target, name) => {
+        if (name === 'window') {
+          return new Proxy(target.window, this[WINDOW]);
+        }
+        return useState(target, name).get();
+      }
+    }
+  }
+
+  get defaultView() {
+    return new Proxy(super.defaultView, this[WINDOW])
+  }
+
+  get location() {
+    return this.defaultView.location;
+  }
+
+  set location(location) {
+    this.defaultView.location = location;
+  }
+
+  get all() { return super.all } 
+  get head() { return super.head } 
+  get body() { return super.body } 
+  get title() { return super.title } 
+  set title(v) { super.title = v } 
```

Comment:
I have no idea why any of these is needed

Reply:
Here's [an example line](https://github.com/WebReflection/linkedom/blob/main/esm/cached.js#L125) that causes an issue during build without `get title()` duplicated. The `Object.getOwnPropertyDescriptors` method does not seem to detect properties from the base class.  Apart from merging `HTMLDocument` and `_HTMLDocument`, the solution would be to avoid calling `getOwnPropertyDescriptors` on `HTMLDocument`.